src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -272,32 +205,48 @@ def fixupProcess(self):
         #   dataset.dataTier
         #   dataset.filterName
         if hasattr(self.process, "outputModules"):
-            outputModuleNames = list(self.process.outputModules)
-        else:
+            outputModuleNames = self.process.outputModules.keys()
```

Comment:
```suggestion
            outputModuleNames = list(self.process.outputModules)
```

In python3 `dict.keys()` returns a [dictionary view object](https://docs.python.org/3.8/library/stdtypes.html#dict-views), that behaves like a generator. In order to have a list in `outputModuleNames`, we have to materialize this generator with `list()`. Eventually, `list(mydict)` is the same as `list(mydict.keys())`, we can use the one that we prefer.